src/CSharp/CodeCracker/Style/SwitchToAutoPropAnalyzer.cs

```diff
@@ -53,12 +52,12 @@ private static void AnalyzeProperty(SyntaxNodeAnalysisContext context, bool canH
             var setterExpressionStatement = setter.Body.Statements.First() as ExpressionStatementSyntax;
             var setterAssignmentExpression = setterExpressionStatement?.Expression as AssignmentExpressionSyntax;
             if (setterAssignmentExpression == null) return;
-            var returnIdentifier = getterReturn.Expression as IdentifierNameSyntax;
+            var returnIdentifier = (getterReturn.Expression is MemberAccessExpressionSyntax ? ((MemberAccessExpressionSyntax)getterReturn.Expression).Name : getterReturn.Expression) as IdentifierNameSyntax;
             if (returnIdentifier == null) return;
             var semanticModel = context.SemanticModel;
             var returnIdentifierSymbol = semanticModel.GetSymbolInfo(returnIdentifier).Symbol;
             if (returnIdentifierSymbol == null) return;
-            var assignmentLeftIdentifier = setterAssignmentExpression.Left as IdentifierNameSyntax;
+            var assignmentLeftIdentifier = (setterAssignmentExpression.Left is MemberAccessExpressionSyntax ? ((MemberAccessExpressionSyntax)setterAssignmentExpression.Left).Name : setterAssignmentExpression.Left) as IdentifierNameSyntax;
```

Comment:
@martinzimmermann On the analyzer you need to check if the `Expression` is `this`. It cannot be any `MemberAccessExpressionSyntax`.
On the code fix it is fine, as it is working on already verified code.